newm-server-common/src/main/kotlin/io/projectnewm/server/ext/StringExt.kt

```diff
@@ -1,5 +1,33 @@
 package io.projectnewm.server.ext
 
+import at.favre.lib.crypto.bcrypt.BCrypt
 import java.util.UUID
 
+// same as [android.util.Patterns.EMAIL_ADDRESS]
+private val EMAIL_REGEX = Regex(
+        "[a-zA-Z0-9\\+\\.\\_\\%\\-\\+]{1,256}" +
+        "\\@" +
+        "[a-zA-Z0-9][a-zA-Z0-9\\-]{0,64}" +
+        "(" +
+        "\\." +
+        "[a-zA-Z0-9][a-zA-Z0-9\\-]{0,25}" +
+        ")+"
+)
+
+private val URL_REGEX = Regex(
+"((http|https)://)(www.)?"
+        + "[a-zA-Z0-9@:%._\\+~#?&//=]"
+        + "{2,256}\\.[a-z]"
+        + "{2,6}\\b([-a-zA-Z0-9@:%"
+        + "._\\+~#?&//=]*)"
+)
+
 fun String.toUUID(): UUID = UUID.fromString(this)
+
+fun String.isValidEmail() = EMAIL_REGEX.matches(this)
+
+fun String.isValidUrl() = URL_REGEX.matches(this)
```

Comment:
nit: I like the convenience of extension functions, but I think that most of these should be top-level functions or scoped to an object. 
They seem to be very specific to particular use-cases as opposed to general string functions